### DISCUSSION TITLE
how to get deep nested json from a query on self relation table in prisma

### DISCUSSION THREAD

**Discussion by danimadmolil** [prisma/prisma#14821] [Q&A] [answered]
i want to create a deep nested menu dynamically with prisma ORM and get all menus with sub menus as a deeply nested json object
this is my code and i already try a query to but the result is not what i want

***this is my prismaSchema file*** 

    model Menu {
      id            Int    @id @default(autoincrement())
      title         String
      url           String @default("#")
      parentMenu    Menu?  @relation("parentSubCategory", fields: [parrentMenuId], references: [id])
      subMenu       Menu[] @relation("parentSubCategory")
      parrentMenuId Int?
    }
i want a query to get a json with deep nested menus and submenus like below

   

     {
         title:"menu1",
         subMenu:[
           {
             title:"sub1"
             subMenus:[
              {
               title:"sub3",
               subMenus:[
               {
                 title:"sub4"
               }
             ]
             } 
           ]  
        }
       ]
        }

I already try query below but the result is not wat i want

    const result = await prisma.menu.findMany({
        select: {
          id: true,
          title: true,
          url: true,
          subMenu: true,
        },
      });

 

**result of my query is like this** 

    "data": [
        {
          "id": 1,
          "title": "sample",
          "url": "#",
          "subMenu": [
            {
              "id": 2,
              "title": "digital",
              "url": "#",
              "parrentMenuId": 1
            }
          ]
        },
        {
          "id": 2,
          "title": "digital",
          "url": "#",
          "subMenu": [
            {
              "id": 3,
              "title": "pc & laptop",
              "url": "#",
              "parrentMenuId": 2
            }
          ]
        },
        {
          "id": 3,
          "title": "pc & laptop",
          "url": "#",
          "subMenu": [
            {
              "id": 4,
              "title": "pc",
              "url": "#",
              "parrentMenuId": 3
            },
            {
              "id": 5,
              "title": "laptop",
              "url": "#",
              "parrentMenuId": 3
            }
          ]
        },
        {
          "id": 4,
          "title": "pc",
          "url": "#",
          "subMenu": [
            {
              "id": 6,
              "title": "pc parts",
              "url": "#",
              "parrentMenuId": 4
            }
          ]
        },
        {
          "id": 5,
          "title": "laptop",
          "url": "#",
          "subMenu": []
        },
        {
          "id": 6,
          "title": "pc parts",
          "url": "#",
          "subMenu": [
            {
              "id": 7,
              "title": "ram and storage",
              "url": "#",
              "parrentMenuId": 6
            }
          ]
        },

can any one help me please?

**Comment by austincrim** [comment, marked as answer]
Hey @danimadmolil 👋,

It's not possible to include the sub-relations to an arbitrary depth in Prisma. You can retrieve a set depth by nesting multiple `select`s or `include`s, but this gets clunky rather quickly.

https://github.com/prisma/prisma/issues/4562 is the related issue and there is some good discussion in there with potential workarounds.

I think your best bet is to use [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw), for now.

**Reply by danimadmolil** [reply]
thanks @austincrim

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
